packages/router-generator/src/generator.ts

```diff
@@ -637,6 +690,51 @@ function removeGroups(s: string) {
   return s.replaceAll(routeGroupPatternRegex, '').replaceAll('//', '/')
 }
 
+/**
+ * Removes the last segment from a given path. Segments are considered to be separated by a '/'.
+ *
+ * @param {string} path - The path from which to remove the last segment. Defaults to '/'.
+ * @returns {string} The path with the last segment removed.
+ * @example
+ * removeLastSegmentFromPath('/workspace/_auth/foo') // '/workspace/_auth'
+ */
+export function removeLastSegmentFromPath(path: string = '/'): string {
+  const segments = path.split('/')
+  segments.pop() // Remove the last segment
+  return segments.join('/')
+}
+
+/**
+ * Removes the last segment from a given variable name. Segments are separated by capital letters.
+ *
+ * @param {string} variableName - The variable name from which to remove the last segment. Defaults to an empty string.
+ * @returns {string} The variable name with the last segment removed.
+ * @example
+ * removeLastSegmentFromVariableName('RouteBBarTanRoute') // 'RouteBBarTan'
+ * removeLastSegmentFromVariableName('RouteAFoo') // 'RouteA'
+ */
+export function removeLastSegmentFromVariableName(
+  variableName: string = '',
+): string {
+  const segments = variableName.split(/(?=[A-Z])/)
```

Comment:
can't we build the variable name again without the last segment instead relying on string manipulations?